keepers/keeper-core/src/lib.rs

```diff
@@ -197,71 +206,11 @@ async fn calculate_instructions_per_tx(
     Ok(size_max.min(compute_max))
 }
 
-async fn parallel_submit_transactions(
-    client: &RpcClient,
-    signer: &Arc<Keypair>,
-    // Each &[Instruction] represents a transaction
-    transactions: &[&[Instruction]],
-    // Map of signature to index of tx in `transactions`
-    executed_signatures: &mut HashMap<Signature, usize>,
-) -> Result<(), TransactionExecutionError> {
-    // Converts arrays of instructions into transactions and submits them in parallel, in batches of 50 (arbitrary, to avoid spamming RPC)
-    // Saves signatures associated with the indexes of instructions it contains
-
-    const TX_BATCH_SIZE: usize = 50;
-    for transaction_batch in transactions.chunks(TX_BATCH_SIZE) {
-        let recent_blockhash = get_latest_blockhash_with_retry(client).await.map_err(|e| {
-            TransactionExecutionError::TransactionClientError(
-                e.to_string(),
-                transactions.iter().map(|&tx| tx.to_vec()).collect(),
-            )
-        })?;
-        // Convert instructions to transactions in batches and send them all, saving their signatures
-        let transactions: Vec<Transaction> = transaction_batch
-            .iter()
-            .map(|batch| {
-                Transaction::new_signed_with_payer(
-                    batch,
-                    Some(&signer.pubkey()),
-                    &[signer.as_ref()],
-                    recent_blockhash,
-                )
-            })
-            .collect();
-
-        let tx_futures = transactions
-            .iter()
-            .enumerate()
-            .map(|(i, tx)| async move {
-                let res = client.send_transaction(tx).await;
-                match res {
-                    Ok(signature) => Some((signature, i)),
-                    Err(e) => {
-                        warn!("Send transaction failed: {:?}", e);
-                        None
-                    }
-                }
-            })
-            .collect::<Vec<_>>();
-
-        let results = futures::future::join_all(tx_futures).await;
-        for (signature, ix_index) in results.into_iter().flatten() {
-            executed_signatures.insert(signature, ix_index);
-        }
-        debug!(
-            "Transactions sent: {}, executed_signatures: {}",
-            transactions.len(),
-            executed_signatures.len()
-        );
-    }
-    Ok(())
-}
-
 async fn parallel_confirm_transactions(
     client: &RpcClient,
-    executed_signatures: &mut HashMap<Signature, usize>,
-) {
-    // Confirmes TXs in batches of 256 (max allowed by RPC method)
+    executed_signatures: HashMap<Signature, usize>,
```

Comment:
nit: you could pass in hashset here then return hashset of confirmed and worry about the index in caller. but this is fine

Reply:
done